lib/backports/3.2.0/enumerator/product.rb

```diff
@@ -47,15 +54,17 @@ def __execute(block, values, enums)
     def size
       total_size = 1
       @__enums.each do |enum|
+        return nil if !enum.respond_to?(:size)
         size = enum.size
-        return size if size == nil || size == Float::INFINITY
+        return size if size == nil || size == Float::INFINITY || size == -Float::INFINITY
+        return nil if !size.kind_of?(Integer) || size.equal?(Float::NAN)
```

Comment:
The NAN condition is redundant, isn't it?

```suggestion
        return nil if !size.kind_of?(Integer)
```

Reply:
Ah yeah I guess it is. I'll fix it.